ER図を作成

y-takagiiii opened the pull request:
ER図を作成しました。レビューお願いします。

## ER図
```mermaid
erDiagram
  Users ||--o{ Posts : ""
  Posts ||--o{ Taggings : ""
  Taggings }o--|| Tags : ""
  Posts ||--|| Videos : ""
  Posts ||--o{ LaughLogs : ""
  Users ||--o{ Favorites : ""
  Posts ||--o{ Favorites : ""
  Users ||--o{ Inquiries : ""
  Inquiries ||--|{ InquiryItems : ""
  Users ||--o{ FollowRelationships : ""
  Users ||--o| ActiveStorageAttachments : ""
  ActiveStorageAttachments ||--|| ActiveStorageBlobs : ""

  Users {
    id integer PK
    name string
    email string
    encrypted_password string
    reset_password_token datetime
    reset_password_sent_at datetime
    remember_created_at datetime
    sign_in_count integer
    current_sign_in_at datetime
    last_sign_in_at datetime
    current_sign_in_ip string
    last_sign_in_ip string
    role integer
    uid string
    provider string
    title integer
    created_at datetime
    updated_at datetime
  }

  Posts {
    id integer PK
    user_id integer FK
    video_id integer FK
    created_at datetime
    updated_at datetime
  }

  Videos {
    id integer PK
    youtube_id string
    title string
    description text
    thumbnail string
    view_count integer
    created_at datetime
    updated_at datetime
  }

  LaughLogs {
    id integer PK
    post_id integer FK
    laugh_time integer
    created_at datetime
    updated_at datetime
  }

  Tags {
    id integer PK
    name string
    created_at datetime
    updated_at datetime
  }

  Taggings {
    id integer PK
    post_id integer FK
    tag_id integer FK
    created_at datetime
    updated_at datetime
  }

  ActiveStorageBlobs {
    id integer PK
    key string
    filename string
    content_type string
    metadata text
    service_name string
    byte_size bigint
    checksum string
    created_at datetime
  }

  ActiveStorageAttachments {
    id integer PK
    name string
    record_type string
    record_id bigint
    blob_id bigint FK
    created_at datetime
  }

  Favorites {
    id integer PK
    user_id integer FK
    post_id integer FK
    created_at datetime
    updated_at datetime
  }

  FollowRelationships {
    id integer PK
    follower_id integer FK
    followed_id integer FK
    created_at datetime
    updated_at datetime
  }

  AdminNotifications {
    id integer PK
    title string
    content text
    created_at datetime
    updated_at datetime
  }

  Inquiries {
    id integer PK
    user_id integer FK
    inquiry_item_id integer FK
    content text
    created_at datetime
    updated_at datetime
  }

  InquiryItems {
    id integer PK
    name string
    created_at datetime
    updated_at datetime
  }
```

## テーブル内容
### users
devise・omniauthの使用を想定
- titleカラム 投稿数に応じてサービス内での称号を獲得できる
### posts
投稿を管理するテーブル
### videos
Youtube Data APIで取得した動画情報を保存するテーブル
### laugh_logs
笑いログを管理するテーブル
"笑いボタン"を押す度に、動画の再生時間(経過時間)を保管、ログ作成時にDBに保存するような実装を想定
- laugh_timeカラム ボタンを押した時間を保存するためのカラム
### tags
タグ名を管理するテーブル
### taggings 
タグ用の中間テーブル
### active_storage_blobs
ユーザーのプロフィール画像を保存するactivestorage用のテーブル
### acteve_storage_attachments
ユーザーのプロフィール画像を保存するactivestorage用のテーブル
### favorites
投稿のお気に入りを管理するテーブル
### follow_relationships
フォロー/フォロワーを管理するテーブル
### admin_notifications 
管理者からの通知を管理するテーブル
### inquiries 
問い合わせ情報を管理するテーブル
### inquiry_items 
問い合わせ項目を管理するテーブル

y-takagiiii committed:
add: ER図を作成

y-takagiiii committed:
fix: ER図を修正

y-takagiiii committed:
add: READMEにER図を追加

y-takagiiii committed:
add: ER図にactivestorage用のテーブルを追加

busitora commented:
- `Taggings` は `PostTags` でも良さそうですが、おまかせします(あまり馴染みない気がする命名な気も?)
- `admin_notifications` ですが、 管理者は users に存在する情報ではないということであってます? 複数管理者がいる場合、 その情報を保持する必要があるので、 `user_id` 持たせてもいいかと思いました。 (`users`に`role` ありますし、そちらが権限管理なのであれば)
- `laugh_time` ですが、  ボタンを押した時間を保存するためのカラム という説明がないと分からなかったので、カラムから推測できる命名だと尚betterだと思いました！

見やすくていいですね！

y-takagiiii committed:
fix: ER図を修正

AdminNotificationsテーブルとUsersテーブルにアソシエーションを追加
TaggingsテーブルをPostTagsテーブルに変更
LaughLogsテーブルのlaugh_timeカラムをbutton_pressed_timeに変更

y-takagiiii committed:
fix: タイポを修正

y-takagiiii commented:
レビューありがとうございます！
以下のように修正しました。ご確認お願いします。
- `Taggings`を`PostTags`に変更しました。
- usersテーブルで権限管理をする想定でしたので、`admin_notifications`に`user_id`を追加して`users`と関連付けました。
  > admin_notifications ですが、 管理者は users に存在する情報ではないということであってます? 複数管理者がいる場合、 その情報を保持する必要があるので、 user_id 持たせてもいいかと思いました。 (usersにrole ありますし、そちらが権限管理なのであれば)
- `laugh_time`を`button_pressed_time`に変更しました。

```mermaid
erDiagram
  Users ||--o{ Posts : ""
  Posts ||--o{ PostTags : ""
  PostTags }o--|| Tags : ""
  Posts ||--|| Videos : ""
  Posts ||--o{ LaughLogs : ""
  Users ||--o{ Favorites : ""
  Posts ||--o{ Favorites : ""
  Users ||--o{ Inquiries : ""
  Inquiries }|--|| InquiryItems : ""
  Users ||--o{ FollowRelationships : ""
  Users ||--o| ActiveStorageAttachments : ""
  ActiveStorageAttachments ||--|| ActiveStorageBlobs : ""
  Users ||--o{ AdminNotifications : ""

  Users {
    id integer PK
    name string
    email string
    encrypted_password string
    reset_password_token datetime
    reset_password_sent_at datetime
    remember_created_at datetime
    sign_in_count integer
    current_sign_in_at datetime
    last_sign_in_at datetime
    current_sign_in_ip string
    last_sign_in_ip string
    role integer
    uid string
    provider string
    title integer
    created_at datetime
    updated_at datetime
  }

  Posts {
    id integer PK
    user_id integer FK
    video_id integer FK
    created_at datetime
    updated_at datetime
  }

  Videos {
    id integer PK
    youtube_id string
    title string
    description text
    thumbnail string
    view_count integer
    created_at datetime
    updated_at datetime
  }

  LaughLogs {
    id integer PK
    post_id integer FK
    button_pressed_time integer
    created_at datetime
    updated_at datetime
  }

  Tags {
    id integer PK
    name string
    created_at datetime
    updated_at datetime
  }

  PostTags {
    id integer PK
    post_id integer FK
    tag_id integer FK
    created_at datetime
    updated_at datetime
  }

  ActiveStorageBlobs {
    id integer PK
    key string
    filename string
    content_type string
    metadata text
    service_name string
    byte_size bigint
    checksum string
    created_at datetime
  }

  ActiveStorageAttachments {
    id integer PK
    name string
    record_type string
    record_id bigint
    blob_id bigint FK
    created_at datetime
  }

  Favorites {
    id integer PK
    user_id integer FK
    post_id integer FK
    created_at datetime
    updated_at datetime
  }

  FollowRelationships {
    id integer PK
    follower_id integer FK
    followed_id integer FK
    created_at datetime
    updated_at datetime
  }

  AdminNotifications {
    id integer PK
    user_id integer FK
    title string
    content text
    created_at datetime
    updated_at datetime
  }

  Inquiries {
    id integer PK
    user_id integer FK
    inquiry_item_id integer FK
    content text
    created_at datetime
    updated_at datetime
  }

  InquiryItems {
    id integer PK
    name string
    created_at datetime
    updated_at datetime
  }
```

busitora commented:
LGTM!!!!!

y-takagiiii commented:
ありがとうございます！
